tests/test_helper.hpp

```diff
@@ -0,0 +1,193 @@
+                  int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(std::complex<double> x, sycl::ext::cplx::complex<double> y,
+                  int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(std::complex<double> x, std::complex<double> y, int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(sycl::ext::cplx::complex<float> x,
+                  sycl::ext::cplx::complex<float> y, int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(sycl::ext::cplx::complex<float> x, std::complex<float> y,
+                  int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(std::complex<float> x, sycl::ext::cplx::complex<float> y,
+                  int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(std::complex<float> x, std::complex<float> y, int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(sycl::ext::cplx::complex<sycl::half> x,
+                  sycl::ext::cplx::complex<sycl::half> y, int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(sycl::ext::cplx::complex<sycl::half> x,
+                  std::complex<sycl::half> y, int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(std::complex<sycl::half> x,
+                  sycl::ext::cplx::complex<sycl::half> y, int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+bool almost_equal(std::complex<sycl::half> x, std::complex<sycl::half> y,
+                  int ulp) {
+  return almost_equal(x.real(), y.real(), ulp) &&
+         almost_equal(x.imag(), y.imag(), ulp);
+}
+
+// Helpers for testing half
+
+std::complex<float> sycl_half_to_float(sycl::ext::cplx::complex<sycl::half> c) {
+  auto c_sycl_float = static_cast<sycl::ext::cplx::complex<float>>(c);
+  return static_cast<std::complex<float>>(c_sycl_float);
+}
+
+std::complex<sycl::half> sycl_float_to_half(std::complex<float> c) {
+  auto c_sycl_half = static_cast<sycl::ext::cplx::complex<sycl::half>>(c);
+  return static_cast<std::complex<sycl::half>>(c_sycl_half);
+}
+
+std::complex<float> trunc_float(std::complex<float> c) {
+  auto c_sycl_half = static_cast<sycl::ext::cplx::complex<sycl::half>>(c);
+  return sycl_half_to_float(c_sycl_half);
+}
+
+template <typename T_in> auto constexpr init_std(T_in re, T_in im) {
```

Comment:
Do you mind renaming `init_std` to `init_std_complex`?  With all those `auto` it was not obvious to me.
Also, do you mind adding a comment to say that we need this function only because of our `emulation` of half-precision / the fact that `std` doesn't have an `half` type

Reply:
Sure, I have update this name and added a comment in `test_helper.hpp`